File: scala/src/main/scala/io/devorbit/sparklabs/batch/sales/rdd/DQDataQuality.scala
package io.devorbit.sparklabs.batch.sales.rdd

object DQDataQuality extends App {
  val duplicates = RddUtils.customers
    .map(customer => (customer.customer_id, 1))
    .reduceByKey(_ + _)
    .filter(_._2 > 1)

  if (duplicates.isEmpty()) println("PASS: customer_id is unique")
  else duplicates.collect().foreach(println)
}

File: python/sparklabs/batch/sales/rdd/dq_data_quality.py
from sparklabs.batch.sales.rdd import utils


def run() -> None:
    duplicates = (
        utils.customers.map(lambda customer: (customer["customer_id"], 1))
        .reduceByKey(lambda left, right: left + right)
        .filter(lambda pair: pair[1] > 1)
    )
    duplicate_rows = duplicates.collect()
    if not duplicate_rows:
        print("PASS: customer_id is unique")
    else:
        for row in duplicate_rows:
            print(row)


if __name__ == "__main__":
    run()
